nilearn/maskers/nifti_labels_masker.py

```diff
@@ -22,12 +22,18 @@ def __init__(self, _resampled_labels_img_, background_label, strategy,
         self.mask_img = mask_img
 
     def __call__(self, imgs):
-        from ..regions import signal_extraction
+        from ..regions.signal_extraction import (
+            _img_to_signals_labels_with_masked_atlas,
+        )
 
-        return signal_extraction.img_to_signals_labels(
-            imgs, self._resampled_labels_img_,
-            background_label=self.background_label, strategy=self.strategy,
-            keep_masked_labels=self.keep_masked_labels, mask_img=self.mask_img)
+        signals, labels, masked_labels_img =\
+            _img_to_signals_labels_with_masked_atlas(
+                imgs, self._resampled_labels_img_,
+                background_label=self.background_label, strategy=self.strategy,
+                keep_masked_labels=self.keep_masked_labels,
+                mask_img=self.mask_img,
+            )
+        return signals, (labels, masked_labels_img)
```

Comment:
And here since this is a private function we can just add the new return without making part of a tuple

Reply:
@ymzayek no, the tuple is because `_filter_and_extract` gets only two inputs and we didn't want to change that.

Reply:
Ok and not wanting to change `_filter_and_extract` is because it is called by other maskers not dealt with in this PR? I think we can leave it like this for this PR but just noting that it seems reasonable to me to change its return behavior if other maskers will be changed to support returning a masked image as well.

Reply:
@ymzayek yes we have to change it's behavior in future